src/xtgeo/grid3d/_grid_wellzone.py

```diff
@@ -127,6 +141,9 @@ def report_zone_mismatch(
 
     res1 = dfuse1["zmatch1"].mean() * 100
 
+    if resultformat == 1:
+        return (res1, mcount1, tcount1)
```

Comment:
Moved this further up in the code to avoid spending time on computing results for format 2 if not needed.